packages/nimble-components/src/table-column/mapping/mappings/icon/index.ts

```diff
@@ -0,0 +1,91 @@
+import {
+    attr,
+    html,
+    observable,
+    ref,
+    ViewTemplate
+} from '@microsoft/fast-element';
+import { DesignSystem } from '@microsoft/fast-foundation';
+import { Mapping } from '..';
+import { styles } from '../styles';
+import { template } from '../template';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-mapping-icon': MappingIcon;
+    }
+}
+
+/**
+ * Maps values to an icon
+ */
+export class MappingIcon extends Mapping {
+    @attr()
+    public icon: string | null = null;
+
+    @attr()
+    public severity: string | null = null;
+
+    @attr()
+    public label: string | null = null;
+
+    /** @internal */
+    public span: HTMLSpanElement | null = null;
+
+    /** @internal */
+    @observable
+    public isValidContentAndHasOverflow = false;
+
+    /** @internal */
+    @observable
+    public cellViewTemplate: ViewTemplate = html``;
+
+    /** @internal */
+    @observable
+    public groupHeaderViewTemplate: ViewTemplate = html``;
+
+    private iconChanged(): void {
+        this.updateCellViewTemplate();
+    }
+
+    private updateCellViewTemplate(): void {
+        if (!this.icon) {
+            return;
+        }
+
+        // prettier-ignore
+        this.cellViewTemplate = html<MappingIcon>`
+            <${this.icon}
```

Comment:
I can't do `${x => x.icon}` because it results in a weird FAST string being used instead of the actual icon tag name. I suspect it is some kind of intermediate symbol that fails to get replaced because it is not bound to an attribute/property. The result is that I have to use `this.icon` instead, and I have to manually update the template whenever the icon changes.